crates/polars-sql/src/functions.rs

```diff
@@ -578,6 +584,19 @@ impl SqlFunctionVisitor<'_> {
             // String functions
             // ----
             EndsWith => self.visit_binary(|e, s| e.str().ends_with(s)),
+            ConcatWs => match function.args.len() {
+                2 => self.try_visit_binary(|e, s: Expr| {
+                    let sep = match e {
+                        Expr::Literal(LiteralValue::Utf8(ref sep)) => sep,
+                        _ => {
+                            polars_bail!(InvalidOperation: "Invalid 'separator' for ConcatWs: {}", function.args[0]);
+                        }
+                    };
+
+                    Ok(s.list().join(sep))
+                }),
+                _ => unimplemented!("Currently `ConcatWs` only works as `concat_ws(sep, col)`")
```

Comment:
can we `polars_bail` instead of panicking here?

Reply:
Is it acceptable if the error value of `InvalidOperation` for now? It can be replaced later if this function gets extended

Reply:
We should always error. Panics are unrecoverable.